libs/pages/clusters/src/lib/ui/page-clusters-create/step-general/step-general.tsx

```diff
@@ -1,30 +1,129 @@
-import { FormEventHandler } from 'react'
-import { useFormContext } from 'react-hook-form'
+import { CloudProvider, CloudProviderEnum, ClusterRegion } from 'qovery-typescript-axios'
+import { FormEventHandler, useEffect, useState } from 'react'
+import { Controller, useFormContext } from 'react-hook-form'
 import { useNavigate, useParams } from 'react-router-dom'
-import { ClusterGeneralSettings } from '@qovery/shared/console-shared'
-import { ApplicationGeneralData } from '@qovery/shared/interfaces'
+import { ClusterCredentialsSettingsFeature, ClusterGeneralSettings } from '@qovery/shared/console-shared'
+import { ClusterGeneralData, Value } from '@qovery/shared/interfaces'
 import { CLUSTERS_URL } from '@qovery/shared/routes'
-import { Button, ButtonSize, ButtonStyle } from '@qovery/shared/ui'
+import {
+  BannerBox,
+  Button,
+  ButtonSize,
+  ButtonStyle,
+  Icon,
+  IconFlag,
+  InputSelect,
+  LoaderSpinner,
+} from '@qovery/shared/ui'
+import { upperCaseFirstLetter } from '@qovery/shared/utils'
 
 export interface StepGeneralProps {
   onSubmit: FormEventHandler<HTMLFormElement>
+  cloudProviders: CloudProvider[]
 }
 
 export function StepGeneral(props: StepGeneralProps) {
-  const { formState } = useFormContext<ApplicationGeneralData>()
+  const { onSubmit, cloudProviders = [] } = props
+  const { control, formState } = useFormContext<ClusterGeneralData>()
   const { organizationId = '' } = useParams()
   const navigate = useNavigate()
 
+  const [currentProvider, setCurrentProvider] = useState<CloudProvider>()
+
+  useEffect(() => {
+    const providerByDefault = cloudProviders?.filter((cloud) => cloud.short_name === CloudProviderEnum.AWS)[0]
+    setCurrentProvider(providerByDefault)
+  }, [cloudProviders])
+
+  const buildCloudProviders: Value[] = cloudProviders.map((value) => ({
```

Comment:
You don't put these values inside a useState or useCallback? Everytime the component rerender you're gonna compute all of that again. It's not the most taxing operation but still better if it's done only when necessary no?